src/plugins/mer/meremulatordevice.cpp

```diff
@@ -280,6 +259,22 @@ void MerEmulatorDevice::executeAction(Core::Id actionId, QWidget *parent)
         return;
     } else if (actionId == Constants::MER_EMULATOR_STOP_ACTION_ID) {
         m_connection->disconnectFrom();
+        return;
+    } else if (actionId == Constants::MER_EMULATOR_CHANGE_MODE_ACTION_ID) {
+        const IDevice::ConstPtr device = DeviceManager::instance()->find(id());
+        const MerEmulatorDevice::Ptr merDevice = device
+                ? device.dynamicCast<const MerEmulatorDevice>().constCast<MerEmulatorDevice>()
+                : sharedFromThis().dynamicCast<MerEmulatorDevice>();
+
+        QTC_ASSERT(merDevice, return);
+
+        MerEmulatorModeDialog dialog(merDevice);
+        const QString deviceModelName = dialog.exec(); // Set device model
+
+        QTC_ASSERT(!deviceModelName.isNull(), return);
+
+        m_deviceModel = deviceModelName; // Set device model in copy of MerEmulatorDevice (in case of Apply pressed)
```

Comment:
WRT to the change suggested to MerEmulatorModeDialog::exec I would change this to

```
if (!dialog.exec())
    return;
// Set device model in copy of MerEmulatorDevice (in case of Apply pressed)
m_deviceModel = merDevice->deviceModel();
```

Reply:
Done.